src/__tests__/asyncHook.fakeTimers.test.ts

```diff
@@ -51,8 +74,273 @@ describe('async hook (fake timers) tests', () => {
 
       expect(complete).toBe(true)
     })
+
+    test('should waitFor arbitrary expectation to pass when fake timers are not advanced explicitly', async () => {
+      const fn = jest.fn().mockReturnValueOnce(false).mockReturnValueOnce(true)
+    
+      const { waitFor } = renderHook(() => null)
+    
+      await waitFor(() => {
+        expect(fn()).toBe(true)
+      })
+    })
+
+    test('should reject if timeout is passed close to when promise resolves', async () => {
+      const { waitFor } = renderHook(() => null)
+
+      let actual = 0
+      const expected = 1
+
+      setTimeout(() => {
+        actual = expected
+      }, 30)
+
+      let complete = false
+
+      await expect(
+        waitFor(
+          () => {
+            expect(actual).toBe(expected)
+            complete = true
+          },
+          { timeout: 29, interval: 10 }
+        )
+      ).rejects.toThrow(Error('Timed out in waitFor after 29ms.'))
+
+      expect(complete).toBe(false)
+    })
+
+    test('should wait for next update', async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useSequence(['first', 'second']))
+
+      expect(result.current).toBe('first')
+
+      await waitForNextUpdate()
+
+      expect(result.current).toBe('second')
+    })
+
+    test('should wait for multiple updates', async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useSequence(['first', 'second', 'third'])
+      )
+
+      expect(result.current).toBe('first')
+
+      await waitForNextUpdate()
+
+      expect(result.current).toBe('second')
+
+      await waitForNextUpdate()
+
+      expect(result.current).toBe('third')
+    })
+
+    test('should reject if timeout exceeded when waiting for next update', async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useSequence(['first', 'second']))
+
+      expect(result.current).toBe('first')
+
+      await expect(waitForNextUpdate({ timeout: 10 })).rejects.toThrow(
+        Error('Timed out in waitForNextUpdate after 10ms.')
+      )
+    })
+
+    // eslint-disable-next-line jest/no-disabled-tests
+    test.skip('should not reject when waiting for next update if timeout has been disabled', async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useSequence(['first', 'second'], 1100))
+
+      expect(result.current).toBe('first')
+
+      await waitForNextUpdate({ timeout: false })
+
+      expect(result.current).toBe('second')
+    })
```

Comment:
this is the one skipped test.  `waitForNextUpdate` has `interval` implicitly set to `false` as well as `timeout`, so without anything advancing timers it' just hangs.  I'm not sure how to deal with this case or if we even should yet.

Reply:
I think I have just found a way to get this case to pass as well.